### PR TITLE
[V1][MLA][SW-234434] Enable MLA for V1 - ported from vllm-gaudi

### PR DESCRIPTION
**Pull request by xuechendi**
https://jira.habana-labs.com/browse/SW-234434

## Essential Elements of an Effective PR Description Checklist
- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results

## Purpose

Backport vllm-gaudi V1 MLA enabling to vllm-fork for PRC customer request

## Test on Deepseek V2 lite chat

```
HABANA_VISIBLE_DEVICES=all VLLM_SKIP_WARMUP=true \
PT_HPU_LAZY_MODE=1 VLLM_USE_V1=1 VLLM_CONTIGUOUS_PA=False \
lm_eval --model vllm   \
             --model_args "pretrained=DeepSeek-V2-Lite-Chat/,tensor_parallel_size=1,distributed_executor_backend=mp,trust_remote_code=true,max_model_len=4096,use_v2_block_manager=True,dtype=bfloat16,max_num_seqs=128"   \
             --tasks gsm8k --num_fewshot "5"   \
             --batch_size "auto"  --log_samples --output_path gsm8k_acc_DeepSeek-V2-Lite-Chat.json
```
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     5|exact_match|↑  |0.6581|±  |0.0131|
|     |       |strict-match    |     5|exact_match|↑  |0.6482|±  |0.0132|

Test on Deepseek R1

```
HABANA_VISIBLE_DEVICES=all \
VLLM_CONTIGUOUS_PA=False \
VLLM_USE_V1=1 \
PT_HPU_LAZY_MODE=1 \
VLLM_SKIP_WARMUP=true \
PT_HPU_ENABLE_LAZY_COLLECTIVES=true \
PT_HPU_WEIGHT_SHARING=0 \
lm_eval --model vllm \
  --model_args "pretrained=DeepSeek-R1,tensor_parallel_size=8,distributed_executor_backend=mp,trust_remote_code=true,max_model_len=16384,use_v2_block_manager=True,dtype=bfloat16,max_num_seqs=128,gpu_memory_utilization=0.9,enable_expert_parallel=True," \
  --tasks gsm8k --num_fewshot "8" \
  --batch_size "128" --limit 256 --log_samples --output_path gsm8k_acc_${MODEL_NAME}.json
```

vllm (pretrained=DeepSeek-R1,tensor_parallel_size=8,distributed_executor_backend=mp,trust_remote_code=true,max_model_len=16384,use_v2_block_manager=True,dtype=bfloat16,max_num_seqs=128,gpu_memory_utilization=0.9,enable_expert_parallel=True,), gen_kwargs: (None), limit: 256.0, num_fewshot: 8, batch_size: 128
|Tasks|Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|-----|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k|      3|flexible-extract|     8|exact_match|↑  |0.9688|±  |0.0109|
|     |       |strict-match    |     8|exact_match|↑  |0.9609|±  |0.0121|

### CODE REVIEW

File: vllm/v1/worker/hpu_model_runner.py [hunk 1]
```diff
@@ -2268,11 +2268,18 @@ def initialize_kv_cache(self, kv_cache_config: KVCacheConfig) -> None:
...
                         num_blocks + 1, kv_cache_spec.block_size,
                         kv_cache_spec.num_kv_heads, kv_cache_spec.head_size)
+                    v_cache_shape = None if self.model_config.use_mla \
+                    else kv_cache_shape
                     dtype = kv_cache_spec.dtype
                     key_cache = torch.zeros(kv_cache_shape,
                                             dtype=dtype,
                                             device=self.device)
-                    value_cache = torch.zeros_like(key_cache)
+                    if v_cache_shape is not None:
```
Comment (michalkuligowski) [inline vllm/v1/worker/hpu_model_runner.py:2277]: You could reverse the condition to test for 'true'
if v_cache_shape is None:
    value_cache = None
else:
    ...

File: vllm/attention/backends/hpu_attn.py [hunk 2]
```diff
@@ -250,18 +250,35 @@ def forward(
...
+            k_cache: torch.Tensor, attn_metadata: HPUAttentionMetadata,
             batch_size: int) -> torch.Tensor:
+        ##### get prefix cache #####
+        if attn_metadata.block_list is not None:
+            current = latent_vec_k
+            past = self.latent_cache_k.fetch_from_cache(
+                k_cache.unflatten(0, (-1, attn_metadata.block_size)),
+                attn_metadata.block_list)
+            past = past.view(-1, past.shape[-1])
+            current = torch.concat((past, current), dim=0)
```
Comment (michalkuligowski) [inline vllm/attention/backends/hpu_attn.py:274]: dim=0 not needed afaik, since dim=0 is default

File: vllm/v1/worker/hpu_model_runner.py (hunk 1 above)
Comment (xuechendi) [reply vllm/v1/worker/hpu_model_runner.py:2277]: @michalkuligowski , you're right, the codes are copied from vllm-gaudi plugin directly, so I want to keep the same coding if it is OK.

File: vllm/attention/backends/hpu_attn.py (hunk 2 above)
Comment (xuechendi) [reply vllm/attention/backends/hpu_attn.py:274]: @michalkuligowski , you're right, the codes are copied from vllm-gaudi plugin directly, so I want to keep the same coding if it is OK.
